### PR TITLE
Project page UI changes

### PR DESCRIPTION
**Pull request by vidya-ram**

### CODE REVIEW

File: funnel/views/project.py [hunk 1]
```diff
@@ -296,15 +296,6 @@ def view_proposals(self):
             'cfp_transition_form': cfp_transition_form,
         }
 
-    @route('videos')
```
Comment (vidya-ram) [inline funnel/views/project.py:299]: Session videos on the main page of project

File: funnel/assets/js/project_header.js [hunk 2]
```diff
@@ -1,7 +1,156 @@
...
+        document.getElementsByTagName('body')[0].appendChild(boxofficeScript);
+      },
+
+      error(response) {
+        const ajaxLoad = this;
+        ajaxLoad.retries -= 1;
+        let errorMsg;
+
+        if (response.readyState === 4) {
+          errorMsg = 'Server error, please try again later.';
```
Comment (jace) [inline funnel/assets/js/project_header.js:46]: gettext needed

Comment (jace) [reply funnel/assets/js/project_header.js:46]: "The server is experiencing difficulties. Try again in a few minutes"

File: funnel/assets/js/project_header.js [hunk 3]
```diff
@@ -1,7 +1,156 @@
...
+          $(widgetElem).html(errorMsg);
+        } else if (response.readyState === 0) {
+          if (ajaxLoad.retries < 0) {
+            if (!navigator.onLine) {
+              errorMsg = 'Unable to connect. There is no network!';
```
Comment (jace) [inline funnel/assets/js/project_header.js:51]: gettext, and "This device has no internet connection"

File: funnel/assets/js/project_header.js [hunk 4]
```diff
@@ -1,7 +1,156 @@
...
+            } else {
+              errorMsg =
+                '<p>Unable to connect. If you are behind a firewall or using any script blocking extension (like Privacy Badger), please ensure your browser can load boxoffice.hasgeek.com, api.razorpay.com and checkout.razorpay.com .</p>';
```
Comment (jace) [inline funnel/assets/js/project_header.js:54]: gettext, and no `<p>` tag inside the translation. Replace "If _you_ are behind" with "If _this device_ is behind"

File: funnel/static/translations/hi_IN/messages.json [hunk 5]
```diff
@@ -1282,6 +1279,11 @@
       "Logging out…": [null, "लॉग आउट हो रहा है…"],
       "Copy link": [null, "लिंक कॉपी करें"],
       "Facebook": [null, "Facebook"],
+      "Edit url": [null, "URL संपादित करें"],
```
Comment (jace) [inline funnel/static/translations/hi_IN/messages.json:1282]: Original should have been "Edit URL" as it's an abbreviation.

File: funnel/static/translations/hi_IN/messages.json [hunk 6]
```diff
@@ -1473,14 +1467,19 @@
         "लिखें, दूसरे से जुड़ें और प्रतिक्रिया पाएं."
       ],
       "Unscheduled projects": [null, "अनिर्धारित प्रोजेक्ट"],
-      "Back to the project": [null, "प्रोजेक्ट पर वापस जाएं"],
-      "Edit details": [null, "विवरण संपादित करें"],
+      "This project will not be listed as it has no sessions in the schedule": [
+        null,
+        ""
+      ],
+      "This project is currently not published. Visit <a href=\"%(url)s\">settings</a> to publish": [
```
Comment (jace) [inline funnel/static/translations/hi_IN/messages.json:1474]: Settings link should not be in the text. It should have an arrow outside the translation, like we use in notifications (where the whole text is a link). Also remove the word "currently" as it's implied when the banner is shown.

CTA in updates feed

File: funnel/static/translations/hi_IN/messages.json [hunk 7]
```diff
@@ -1509,9 +1508,54 @@
       ],
       "Open now": [null, "अभी खोलें"],
       "Open later": [null, "बाद में खोलें"],
+      "This project is brought to you by": [null, ""],
```
Comment (jace) [inline funnel/static/translations/hi_IN/messages.json:1511]: Where is this from? It seems like it's cut mid sentence.

File: funnel/templates/project_layout.html.jinja2 [hunk 8]
```diff
@@ -0,0 +1,97 @@
+{% extends "layout.html.jinja2" %}
+{% set title_suffix = project.title_suffix %}
+{% from "baseframe/forms.html.jinja2" import ajaxform %}
+{% from "baseframe/components.html.jinja2" import faicon %}
+{%- from "macros.html.jinja2" import csrf_tag, project_header, project_details_header, proposal_card, project_share, video_thumbnail, registerblock, project_col_right %}
+{% block title %}{{ project.title }}{% endblock %}
+{% block description %}{{ project.tagline }}{% if project.tagline %} &ndash; {{ project.taglines }}{% endif %}{% endblock %}
```
Comment (jace) [inline funnel/templates/project_layout.html.jinja2:7]: Why do we have `tagline` and `taglines`? It's confusing to have two similarly named attributes.

File: funnel/templates/project_membership.html.jinja2 [hunk 9]
```diff
@@ -1,52 +1,22 @@
...
-{% block basecontent %}
-  <div class="mui-container">
-    <div class="page-content page-content--mob-nav">
-      <div class="grid" id="crew">
-        <div class="grid__col-xs-12">
-          {{ membership_template() }}
-        </div>
-      </div>
-    </div>
+{% block basecontentinner %}
```
Comment (jace) [inline funnel/templates/project_membership.html.jinja2:9]: Some of these names are getting large. Maybe break up with underscores, like `base_content_inner`? Also, these blocks are getting spaghetti-like and will need some refactoring later.

File: funnel/templates/macros.html.jinja2 [hunk 10]
```diff
@@ -335,6 +364,21 @@
...
+    {% if not project.schedule_state.PAST and project.features.tickets() -%}
+      <div class="tickets-wrapper__modal">
+        <div class="mui--hidden-md mui--hidden-lg mui--hidden-xl tickets-wrapper__modal__back">
+          <div class="mobile-nav mui--z1">
+            <a href="javascript:void(0)" aria-label="{% trans %}Back to the project{% endtrans %}" class="mui--text-dark mobile-nav__icon" id="close-ticket-widget" data-ga="Back to the project page">{{ faicon(icon='arrow-left', icon_size='title') }}</a><span class="mui--text-dark mobile-nav__headline">{% trans %}Tickets{% endtrans %}</span>
+          </div>
+        </div>
+        <div id="tickets" class="tickets-wrapper__modal__body">
+          <a href="javascript:void(0)" aria-label="{% trans %}Back to the project{% endtrans %}" class="tickets-wrapper__modal__body__close" id="close-ticket-widget" data-ga="Back to the project page">{{ faicon(icon='times', icon_size='title') }}</a>
+          <div id="boxoffice-widget"><p class="mui--text-body2">Loading...</p></div>
```
Comment (jace) [inline funnel/templates/macros.html.jinja2:377]: Needs `{% trans %}` and `…` instead of `...`

File: funnel/templates/project.html.jinja2 [hunk 11]
```diff
@@ -58,259 +52,148 @@
...
-        </div>
-      </div>
+{% block basecontentinner %}
+  {% if project.view_for('edit').is_available() and project.features.schedule_no_sessions() %}
+    {{ alertbox('warning', gettext("This project will not be listed as it has no sessions in the schedule")) }}
+  {% endif %}
+  {% if project.view_for('view').is_available() and project.state.DRAFT %}
+    <div class="alert alert--warning alert--dismissable">
+      <a class="alert__close" href="javascript:void(0);" aria-label="{% trans %}Close{% endtrans %}">{{ faicon(icon='times', icon_size='title', baseline=true) }}</a>
+      <p class="alert__text">{% trans url=project.url_for('settings') %}This project is currently not published. Visit <a href="{{ url }}">settings</a> to publish{% endtrans %}</p>
```
Comment (jace) [inline funnel/templates/project.html.jinja2:75]: As above. The link should be for the entire text, outside the translation.

**Review by jace — CHANGES_REQUESTED** [review]

File: funnel/static/translations/hi_IN/messages.json (hunk 7 above)
Comment (vidya-ram) [reply funnel/static/translations/hi_IN/messages.json:1511]: Screen Shot 2021-02-12 at 10 08 15 AM

File: funnel/templates/project_layout.html.jinja2 (hunk 8 above)
Comment (vidya-ram) [reply funnel/templates/project_layout.html.jinja2:7]: Typo error. Fixed it

**Pull request merged**